### PR TITLE
update bit type

### PR DESCRIPTION
**Pull request by siddontang**
+ handle bit field length in parser
+ use Bit type instead of old uint64
+ return bit type binary string for mysql protocol

### CODE REVIEW

File: util/types/convert.go [hunk 1]
```diff
@@ -386,10 +386,8 @@ func Convert(val interface{}, target *FieldType) (v interface{}, err error) { //
 		// check bit boundary, if bit has n width, the boundary is
 		// in [0, (1 << n) - 1]
 		width := target.Flen
-		if width == 0 {
+		if width == 0 || width == mysql.UnspecifiedBitWidth {
```
Comment (qiuyesuifeng) [inline util/types/convert.go]: If `width == mysql.UnspecifiedBitWidth` use mysql.MinBitWidth?

Comment (siddontang) [inline util/types/convert.go]: yes, default length is 1 for mysql bit.

**Comment by qiuyesuifeng** [comment]
LGTM

**Comment by coocood** [comment]
LGTM

**Pull request merged**